system_tests/block_validator_test.go

```diff
@@ -78,15 +82,66 @@ func testBlockValidatorSimple(t *testing.T, opts Options) {
 		builder.WithArbOSVersion(opts.arbosVersion)
 	}
 	builder.L2Info = nil
-	cleanup := builder.Build(t)
-	defer cleanup()
 
-	authorizeDASKeyset(t, ctx, dasSignerKey, builder.L1Info, builder.L1.Client)
+	// Configure for referenceda mode - deploy validator contract
+	if opts.dasModeString == "referenceda" {
+		builder.WithReferenceDA()
+	}
+
+	// For ReferenceDA, we need to build L1 first, create the external provider server,
+	// then build L2. For other DA modes, we can build in one go.
+	var refDAURL string
+	var refDAServer *http.Server
+	if opts.dasModeString == "referenceda" {
```

Comment:
I'm still reading the tests, but I wonder - why can't we put all this server deployment logic into the builder (like we did for OSP contracts deployment)? It feels like this is pretty a universal setup for ReferenceDA